src/test/java/org/broadinstitute/dsm/util/tools/TbosUPSKitTool.java

```diff
@@ -0,0 +1,510 @@
+                for (; i < headers.length; ) {
+                    String header = headers[i];
+                    String value = participantInfo[i];
+                    if (StringUtils.isBlank(value) || "-".equals(value)) {
+                        value = null;
+                    }
+                    if ("kit label".equals(header)) {
+                        if (kit != null && !kit.isEmpty() && StringUtils.isNotBlank(kit.getKitLabel())) {
+                            if (!participants.containsKey(shortId)) {
+                                participants.put(shortId, new ArrayList<Kit>());
+                            }
+                            ArrayList<Kit> list = participants.get(shortId);
+                            list.add(kit);
+                            participants.put(shortId, list);
+                            kit = null;
+                        }
+                        else if (kit != null && StringUtils.isBlank(kit.getKitLabel())) {
+                            kit = null;
+                        }
+                    }
+                    if (kit == null) {
+                        kit = new Kit();
+                        kit.setShortId(shortId);
+                        kit.setGuid(guid);
+                    }
+                    kit.setValueByHeader(header, value);
+                    i++;
+                }
+            }
+            if (kit != null && !kit.isEmpty() && StringUtils.isNotBlank(shortId)) {// last kit of the participant in the last line
+                if (!participants.containsKey(shortId)) {
+                    participants.put(shortId, new ArrayList<Kit>());
+                }
+                ArrayList<Kit> list = participants.get(shortId);
+                list.add(kit);
+                participants.put(shortId, list);
+                kit = null;
+            }
+
+        }
+        catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        return participants;
+    }
+
+    public static void insertFileInfo(Map<String, ArrayList<Kit>> participants) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstance("testboston");
+        String[] shortIds = participants.keySet().toArray(new String[0]);
+        int counter = 0;
+        for (; counter < shortIds.length; counter++) {
+            String participantShortId = shortIds[counter];
+            logger.info("starting on partcipant " + (counter+1) + "/" + shortIds.length);
+            ArrayList<Kit> kits = participants.get(participantShortId);
+            logger.info("Found " + kits.size() + " kits for the particpant in this file ");
+            String insertedShipmentId = null;
+            int kitCounter = 0;
+            for (; kitCounter < kits.size(); kitCounter++) {
+                Kit kit = kits.get(kitCounter);
+                Kit outboundKit = getDDBKitBasedOnKitLabel(SQL_SELECT_KIT_BY_KIT_LABEL + SQL_SELECT_OUTBOUND, kit.getKitLabel(), ddpInstance.getDdpInstanceId());
```

Comment:
Find the kit info in the DB, so we can find the tracking numbers and also see if there is already activity info for the kit in db